recipe-client-addon/test/browser/browser_NormandyDriver.js

```diff
@@ -90,3 +90,69 @@ add_task(withSandboxManager(Assert, async function testCreateStorage(sandboxMana
     })();
   `);
 }));
+
+add_task(withDriver(Assert, async function addons(driver) {
+  const ADDON_ID = "normandydriver@example.com";
+  const startDate = new Date();
+  const dir = getChromeDir(getResolvedURI(gTestPath));
+  dir.append("fixtures");
+  dir.append("normandy.xpi");
+  const xpiUrl = Services.io.newFileURI(dir).spec;
+
+  let addon = await driver.addons.get(ADDON_ID);
+  Assert.equal(addon, null, "Add-on is not yet installed");
+
+  await driver.addons.install(xpiUrl);
+  addon = await driver.addons.get(ADDON_ID);
+
+  Assert.notEqual(addon, null, "Add-on object was returned");
+  // compare with seconds-level accuracy
+  Assert.greaterOrEqual(Math.floor(addon.installDate / 1000), Math.floor(startDate / 1000), "Add-on is installed in the past.");
+  Assert.less(addon.installDate.valueOf() - startDate.valueOf(), 60 * 1000, "Add-on is installed recently.");
+
+  Assert.deepEqual(addon, {
+    id: "normandydriver@example.com",
+    name: "normandy_fixture",
+    version: "1.0",
+    installDate: addon.installDate,
+    isActive: true,
+    type: "extension",
+  }, "Add-on is installed");
+
+  await driver.addons.uninstall(ADDON_ID);
+  addon = await driver.addons.get(ADDON_ID);
+
+  Assert.equal(addon, null, "Add-on has been uninstalled");
+}));
+
+add_task(withSandboxManager(Assert, async function testAddonsGetWorksInSandbox(sandboxManager) {
```

Comment:
I tried to make a test helper for this, and it really ended up being more trouble than it was worth for now.